// @dcl/schemas/dist/platform/scene/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpawnPoint = exports.Source = exports.RequiredPermission = exports.Scene = exports.SceneParcels = exports.WorldConfiguration = exports.FeatureToggles = void 0;
var feature_toggles_1 = require("./feature-toggles");
Object.defineProperty(exports, "FeatureToggles", { enumerable: true, get: function () { return feature_toggles_1.FeatureToggles; } });
var world_configuration_1 = require("./world-configuration");
Object.defineProperty(exports, "WorldConfiguration", { enumerable: true, get: function () { return world_configuration_1.WorldConfiguration; } });
var scene_parcels_1 = require("./scene-parcels");
Object.defineProperty(exports, "SceneParcels", { enumerable: true, get: function () { return scene_parcels_1.SceneParcels; } });
var scene_1 = require("./scene");
Object.defineProperty(exports, "Scene", { enumerable: true, get: function () { return scene_1.Scene; } });
Object.defineProperty(exports, "RequiredPermission", { enumerable: true, get: function () { return scene_1.RequiredPermission; } });
var source_1 = require("./source");
Object.defineProperty(exports, "Source", { enumerable: true, get: function () { return source_1.Source; } });
var spawn_point_1 = require("./spawn-point");
Object.defineProperty(exports, "SpawnPoint", { enumerable: true, get: function () { return spawn_point_1.SpawnPoint; } });
